// react-moment-datepicker/dist/Day.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _reactMomentProptypes = require('react-moment-proptypes');

var _reactMomentProptypes2 = _interopRequireDefault(_reactMomentProptypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('./Day.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Day = function Day(_ref) {
    var date = _ref.date;
    var onClick = _ref.onClick;
    var displayedStartDate = _ref.displayedStartDate;
    var isSelected = _ref.isSelected;
    var isDefault = _ref.isDefault;


    var classes = (0, _classnames2.default)('rm_day', {
        'rm_day--outside': date.month() !== displayedStartDate.month(),
        'rm_day--past': date.diff((0, _moment2.default)().startOf('day'), 'day') < 0,
        'rm_day--selected': isSelected,
        'rm_day--default': isDefault
    });

    return _react2.default.createElement(
        'div',
        { onClick: onClick, className: classes },
        _react2.default.createElement(
            'button',
            { className: 'rm_day-label' },
            date.date()
        )
    );
};

Day.PropTypes = {
    date: _reactMomentProptypes2.default.momentObj,
    onClick: _react2.default.PropTypes.func,
    displayedStartDate: _reactMomentProptypes2.default.momentObj,
    isSelected: _react2.default.PropTypes.bool,
    isDefault: _react2.default.PropTypes.bool
};

exports.default = Day;
module.exports = exports['default'];
